americangut/util.py

```diff
@@ -285,3 +366,109 @@ def count_samples(metadata_fp, criteria=None):
             count += 1
 
     return count
+
+
+simple_matter_map = {
+    'feces': 'FECAL',
+    'sebum': 'SKIN',
+    'tongue': 'ORAL',
+    'skin': 'SKIN',
+    'mouth': 'ORAL',
+    'gingiva': 'ORAL',
+    'gingival epithelium': 'ORAL',
+    'nares': 'SKIN',
+    'skin of hand': 'SKIN',
+    'hand': 'SKIN',
+    'skin of head': 'SKIN',
+    'hand skin': 'SKIN',
+    'throat': 'ORAL',
+    'auricular region zone of skin': 'SKIN',
+    'mucosa of tongue': 'ORAL',
+    'mucosa of vagina': 'SKIN',
+    'palatine tonsil': 'ORAL',
+    'hard palate': 'ORAL',
+    'saliva': 'ORAL',
+    'stool': 'FECAL',
+    'vagina': 'SKIN',
+    'fossa': 'SKIN',
+    'buccal mucosa': 'ORAL',
+    'vaginal fornix': 'SKIN',
+    'hair follicle': 'SKIN',
+    'nostril': 'SKIN'
+}
+
+
+def clean_and_reformat_mapping(in_fp, out_fp, body_site_column_name,
+                               exp_acronym):
+    """Simplify the mapping file for use in figures
+
+    in_fp : input file-like object
+    out_fp : output file-like object
+    body_site_column_name : specify the column name for body
+    exp_acronym : short name for the study
+
+    Returns a dict containing a description of any unprocessed samples.
+    """
+    errors = defaultdict(list)
+
+    mapping_lines = [l.strip('\n').split('\t') for l in in_fp]
+
+    header = mapping_lines[0]
+    header_low = [x.lower() for x in header]
+
+    bodysite_idx = header_low.index(body_site_column_name.lower())
+    country_idx = header_low.index('country')
+
+    new_mapping_lines = [header[:]]
+    new_mapping_lines[0].append('SIMPLE_BODY_SITE')
+    new_mapping_lines[0].append('TITLE_ACRONYM')
+    new_mapping_lines[0].append('TITLE_BODY_SITE')
+    new_mapping_lines[0].append('HMP_SITE')
+
+    for l in mapping_lines[1:]:
+        new_line = l[:]
+        sample_id = new_line[0]
+        body_site = new_line[bodysite_idx]
+        country = new_line[country_idx]
+
+        # grab the body site
+        if body_site.startswith('UBERON_'):
+            body_site = body_site.split('_', 1)[-1].replace("_", " ")
+        elif body_site.startswith('UBERON:'):
+            body_site = body_site.split(':', 1)[-1]
+        elif body_site in ['NA', 'unknown', '', 'no_data']:
```

Comment:
```python
In [1]: x = ['string1', 'string2', 'string3', 'string4']

In [2]: y = {'string1', 'string2', 'string3', 'string4'}

In [3]: timeit lambda r: r in x
The slowest run took 20.53 times longer than the fastest. This could mean that an intermediate result is being cached 
10000000 loops, best of 3: 58.1 ns per loop

In [4]: timeit lambda r: r in y
The slowest run took 32.97 times longer than the fastest. This could mean that an intermediate result is being cached 
10000000 loops, best of 3: 57.9 ns per loop
```
Probably not, but is good for consistency.